engine/access/rest/request_decorator.go

```diff
@@ -0,0 +1,145 @@
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+
+	"github.com/onflow/flow-go/engine/access/rest/middleware"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// a convenience wrapper around the http request to make it easy to read request query params
+type request struct {
+	*http.Request
+	expandFields map[string]bool
```

Comment:
nit: by convention, a map which represents a set should have type `map[string]struct{}` in Go. This is to prevent ambiguity in the case where a bool is set to false.

Reply:
went with the `map[string]bool` since its it a little easier to reader `if map[key] { ` then the if _, ok = map[key]; ok` pattern.